plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,311 @@
+        cmd.append("--identity=-")  # Read the user record from standard input.
+        return(self.module.run_command(cmd, data=record, use_unsafe_shell=True))
+
+    def _hash_password(self, password):
+        # TODO handle errors etc.
+        return(crypt.crypt(password))
+
+    def remove_user(self):
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append('remove')
+        cmd.append(self.name)
+        return(self.module.run_command(cmd, use_unsafe_shell=True))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            name=dict(type='str', required=True, aliases=['user', 'username']),
+            password=dict(type='str', no_log=True, required=True),
+            storage=dict(type='str', default='classic', choices=['classic', 'luks', 'directory', 'subvolume', 'fscrypt', 'cifs']),
+            disksize=dict(type='str'),
+            realname=dict(type='str', aliases=['real_name']),
+            realm=dict(type='str'),
+            email=dict(type='str'),
+            location=dict(type='str'),
+            iconname=dict(type='str'),
+            homedir=dict(type='path'),
+            uid=dict(type='int'),
+            gid=dict(type='int'),
+            umask=dict(type='int'),
+            environment=dict(type='list', elements='str'),
+            timezone=dict(type='str'),
+            member=dict(type='list', elements='str'),
+            shell=dict(type='str', default='/bin/bash'),
+        ),
+        supports_check_mode=True,
+    )
+
+    homectl = Homectl(module)
+    rc = None
+    result = {}
+    result['name'] = homectl.name
+    result['state'] = homectl.state
+
+    # First we need to make sure homed service is active
+    if not homectl.homed_service_active():
+        module.fail_json(msg="systemd-homed.service is not active")
+
+    # handle removing user
+    if homectl.state == 'absent':
+        if homectl.user_exists():
+            if module.check_mode:
+                module.exit_json(changed=True)
+            rc, stdout, stderr = homectl.remove_user()
+            if rc != 0:
+                module.fail_json(name=homectl.name, msg=stderr, rc=rc)
+            result['changed'] = True
+            result['msg'] = "User %s removed!" % homectl.name
+        else:
+            result['msg'] = "User Doesn't Exist!"
+            result['changed'] = False
+
+    # Handle adding a user
+    if homectl.state == 'present':
+        if not homectl.user_exists():
+            if module.check_mode:
+                module.exit_json(changed=True)
+            rc, stdout, stderr = homectl.create_user()
+            if rc != 0:
+                module.fail_json(name=homectl.name, msg=stderr, rc=rc)
+            result['changed'] = True
+            result['msg'] = "User %s created!" % homectl.name
+        else:
+            result['msg'] = "User already Exists!"
```

Comment:
If the module does no idempotency checks except existence, this needs to be clearly documented.

Reply:
Can you elaborate? When a user is to be added or removed this check will happen. I plan to add more code to modify a current user which will also check of user exists and grabs the current metadata for that user.

Reply:
I think I see what you are saying.

So here:

```
else:
     result['msg'] = "User already Exists!"
```

should also make sure the "chaged" would be False such as:

```python3
else:
      result['changed'] = False
      result['msg'] = "User already Exists!"
```

Reply:
Yes, it should set `changed=False`, **but** the module documentation should also say that no changes are made when `state=present`. This is different from most other Ansible modules. (Your `modify` state should be a part of `present`.)